lib/iris/aux_factory.py

```diff
@@ -784,20 +758,51 @@ def dependencies(self):
                     depth_c=self.depth_c, nsigma=self.nsigma, zlev=self.zlev)
 
     def _derive(self, sigma, eta, depth, depth_c,
-                zlev, shape, nsigma_slice):
+                zlev, nsigma, coord_dims_func):
+        # Calculate the index of the 'z' dimension in the inputs.
+        # Get the cube dimension...
+        i_levels_cubedim, = coord_dims_func(self.dependencies['zlev'])
+        # ...subtract any previous dimensions not used in the calculation.
+        derived_cubedims = self.derived_dims(coord_dims_func)
+        i_levels_dim = i_levels_cubedim - sum(
+            i_dim not in derived_cubedims
+            for i_dim in range(i_levels_cubedim))
```

Comment:
That naming was an attempt to distinguish between the dependency arguments, which are all arrays of the same dimensionality, and the "extra" args which as ordinary Python values.
Hence the "i_" -- it is not an indirection.

The problem is, we have two contexts for 'dimension' or 'index' : The original cube, and the dependency arguments.  That's why I need to calculate 'i_levels_dim' from 'i_levels_cubedim'.

I'll think on ...